### PR TITLE
moved CrossColumnMappingTransformer tests to new approach

### PR DESCRIPTION
**Pull request by limlam96**
Issues: [Base Class](https://github.com/lvgig/tubular/issues/214), [main class](https://github.com/lvgig/tubular/issues/215)

Created BaseCrossColumnMappingTransformer class with functionality shared throughout classes in file, integrated this into new testing approach

### CODE REVIEW

File: tests/mapping/test_BaseMappingTransformer.py [hunk 1]
```diff
@@ -110,7 +115,13 @@ def test_mappings_unchanged(self, uninitialized_transformers):
             "b": {"a": 1, "b": 2, "c": 3, "d": 4, "e": 5, "f": 6},
         }
 
-        x = uninitialized_transformers[self.transformer_name](mappings=mapping)
+        if self.transformer_name == "CrossColumnMappingTransformer":
```
Comment (davidhopkinson26) [inline tests/mapping/test_BaseMappingTransformer.py:118]: Not sure I like having conditional splits in logic for different transformers within a test like this.  Might be clearer and less coupled if specific behaviour for children is specified and tested in its own test module. Can we just overwrite the test for CrossColumnMappingTransformer? Appreciate that means duplicating code but would like to keep the logical flow of these tests consentient.

File: tubular/mapping.py [hunk 2]
```diff
@@ -217,7 +217,70 @@ def transform(
         return X
 
 
-class CrossColumnMappingTransformer(BaseMappingTransformer):
+class BaseCrossColumnMappingTransformer(BaseMappingTransformer):
```
Comment (davidhopkinson26) [inline tubular/mapping.py:220]: P - good shout on pulling this out into a base class.  A case could be made for creating a new sub module for cross column transformers but for now I think this is fine.  (and these test updates hopefully make such a refactor very straightforward in future)

**Review by davidhopkinson26 — CHANGES_REQUESTED** [review]
Looks good, changes requested to `test_mappings_unchanged`.  Let me know what you think.

**Comment by davidhopkinson26** [comment]
please also update the CHANGELOG

File: tubular/mapping.py (hunk 2 above)
Comment (limlam96) [reply tubular/mapping.py:220]: Thanks David :) agree could be worth splitting in future

**Comment by limlam96** [comment]
changelog updated [here](https://github.com/lvgig/tubular/pull/236/commits/030f0b32cbdb5076b731374ba364426c1de3dc92)

File: tests/mapping/test_BaseMappingTransformer.py (hunk 1 above)
Comment (limlam96) [reply tests/mapping/test_BaseMappingTransformer.py]: ah overwriting/overloading makes sense thanks, updated [here](https://github.com/lvgig/tubular/pull/236/commits/bbffa2b5b196e12d92565f0500c3f81c79479b0d)

**Review by davidhopkinson26 — APPROVED** [review]
Looks good

**Pull request merged**